webapp/components/post_view/components/new_message_indicator.jsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+export default class NewMessageIndicator extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            visible: false,
+            rendered: false
+        };
+    }
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.newMessages > 0) {
+            this.setState({rendered: true}, () => {
```

Comment:
For example, if the newMessages prop is >0 the component is rendered and visible to the user.
When the prop changes to 0, the component will animate out of screen. This is done by adding a class that will make sure the component will smoothly disappear.  But when the animation is finished, the component is still _there_, its just not visible for the user because of it's offset.
While developing I encountered a case where animating out of the screen wasn't always enough. This is possible because the component _assumes_ 30px down is _offscreen_. this however, can easily change in the future by some minor change. 
To make the component more resilient to these kinds of changes, I decided to set ```display:none``` after the animation is finished to make sure the whole component is not rendered unless it's really necessary.

It's a similair technique as used by the [animation add-ons](https://facebook.github.io/react/docs/animation.html) in react. This addon allows these kinds of techniques to be applied automatically based on hooks like ```ComponentDidLeave() and componentDidAppear()```